### PR TITLE
Add sanity check and complement PR #139 with tests (add history)

### PR DESCRIPTION
**Pull request by jimmiehansson**
**Description**
This PR complements existing PR #139 with tests and a solution for filtering primitives and parsing numbers properly.

* Add number-to-string coercion for ignored keys
* Exclude null references for ignored keys
* Add tests

**OBS:** This PR renders #144 obsolete as it does not contain the proper git history and attribution to @rohit-gohri

Pardon the chaotic history of this PR

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -307,8 +307,16 @@ function prettifyObject ({
 
   let result = ''
 
-  const keysToIterate = objectKeys.filter(k => keysToIgnore.includes(k) === false)
-  for (var i = 0; i < objectKeys.length; i += 1) {
+  function safeObjectKeys () {
```
Comment (jsumners) [inline lib/utils.js:310]: Could we move this declaration below `return result`? It'll be hoisted appropriately and I think the organization is easier to read.

Comment (jimmiehansson) [reply lib/utils.js]: Sure, that makes sense. I've updated the method placement.

**Comment by rohit-gohri** [comment]
Thanks @jimmiehansson for taking this up and sorry I didn't end up adding tests to the original PR, was in the middle of changing jobs.

File: lib/utils.js [hunk 2]
```diff
@@ -307,8 +307,16 @@ function prettifyObject ({
...
+    const safePrims = objectKeys.filter(k => k !== 'null')
+    const strArr = keysToIgnore.join(',').split(',')
+    const sanitized = safePrims.filter(k => strArr.includes(k) === false)
+    return sanitized
+  }
+
+  const keysToIterate = safeObjectKeys()
+  var i = 0
+  for (i; i < objectKeys.length; i += 1) {
```
Comment (rohit-gohri) [inline lib/utils.js:312]: I couldn't help but notice that this code is again iterating `objectKeys` array but using the index on `keysToIterate` variable. Shouldn't this be :

```suggestion
  for (i; i < keysToIterate.length; i += 1) {
```

Comment (jimmiehansson) [reply lib/utils.js]: Thanks for catching this Rohit, this was an unintended reverse during branch changing

**Review by mcollina — APPROVED** [review]
lgtm

**Review by jsumners — APPROVED** [review]
LGTM

Just need to update the branch.

File: lib/utils.js [hunk 3]
```diff
@@ -349,6 +350,18 @@ function prettifyObject ({
...
+   * Should return a sanitized list of ignored keys
+   *
+   * @returns {iterable}
+   */
+  function safeObjectKeys () {
+    const safePrims = objectKeys.filter(k => k !== 'null')
+    const strArr = keysToIgnore.join(',').split(',')
+    const sanitized = safePrims.filter(k => strArr.includes(k) === false)
+    return sanitized
+  }
```
Comment (mcollina) [inline lib/utils.js:364]: can you move this out of the main function:

```js
  function safeObjectKeys (objectKeys, strArray) {
    const safePrims = objectKeys.filter(k => k !== 'null')
    const sanitized = safePrims.filter(k => strArr.includes(k) === false)
    return sanitized
  }
```

**Review by mcollina — CHANGES_REQUESTED** [review]

**Comment by mcollina** [comment]
There is also a conflict to fix.

**Comment by mcollina** [comment]
Close with no activity.

**Pull request closed without merging**